lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -1121,11 +1142,23 @@ static ParseResult parseFModuleLikeOp(OpAsmParser &parser,
 }
 
 ParseResult FModuleOp::parse(OpAsmParser &parser, OperationState &result) {
-  return parseFModuleLikeOp(parser, result, /*hasSSAIdentifiers=*/true);
+  if (parseFModuleLikeOp(parser, result, /*hasSSAIdentifiers=*/true))
+    return failure();
+  if (!result.attributes.get("convention"))
```

Comment:
This is how this is done elsewhere, but consider hoisting the attribute name to some global location.